packages/nextjs/pages/index.tsx

```diff
@@ -269,4 +280,17 @@ const Home: NextPage = () => {
   );
 };
 
+export const getStaticProps: GetStaticProps<{ stats: Stats }> = async () => {
+  const res = await fetch("https://buidlguidl-v3.ew.r.appspot.com/api/stats");
+  const stats = (await res.json()) as Stats;
```

Comment:
Could you wrap this with try / catch?

Reply:
Such a rookie mistake I missed it my bad 🤦‍♂️

Actually, I was thinking about how to handle error nicely... maybe return all the values as `0` if something fails but then I found this :  

> If there is an error inside getStaticProps when handling background regeneration, or you manually throw an error, the last successfully generated page will continue to show. On the next subsequent request, Next.js will retry calling getStaticProps.
 > [Nextjs docs error handling ISR](https://nextjs.org/docs/pages/building-your-application/data-fetching/incremental-static-regeneration#error-handling-and-revalidation)
 
Since we are already using `reValidate` we have opted in for ISR. So hopefully it should show last-fetched stats if something fails which I think is great instead of showing all 0 values or showing the Error / PageNotFound page 

Also, I stumbled upon this discussion https://github.com/vercel/next.js/discussions/11180#discussioncomment-2399

But please feel free to correct me if I missed something,  Will update 🙌

Reply:
Yes! That makes a lot of sense. It also depends on how we want to handle it.

> So hopefully it should show last-fetched stats if something fails which I think is great instead of showing all 0 values or showing the Error / PageNotFound page

:100: !

With ISR:
- If it fails on build time => the build won't complete
- If it fails on "SSR" request when refreshing after 6 hours... it'll show the previous version.  

The only reason to try/catch is maybe throwing our custom error message, so it's easy to spot on the logs.

But the way you did it with the response, it's more than enough!

Great stuff, thanks!